### ISSUE TITLE
Reopen the run-windows at the last known position

### ISSUE THREAD

**Issue by ngreve** [godotengine/godot-proposals#4900]
### Describe the project you are working on

Godot Games

### Describe the problem or limitation you are having in your project

Many developers do work with some sort of multi monitor setup, including me.
The setup and workflow is mostly as the following:
 - When working with Godot, the editor is usually running in full screen on one monitor. At some point you want to test what you've put together so you it the run-button.
 - The game window opens on top of the editor in the middle of the screen.
 - You find something in your game that does not work as expected and you move the game window onto your second screen.
 - You change some things in your game, watching the run-window on your seconds screen. At some point you hit the (re)run-button.
 - The run window opens in the middle of the screen again.

I explained it with the most useful desktop setup, but this improvement would also be very useful when working with 1 monitor. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

When hitting the (re)run button, the game window should reopen at the position it was put before.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![godot-proposal](https://user-images.githubusercontent.com/18660100/179408365-64b29a5f-d63e-406a-bf3e-c74122789abe.jpg)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't think so (?)

### Is there a reason why this should be core and not an add-on in the asset library?

Not an engine developer, but to me it just makes sense and would be overkill for this **seemingly** simple feature/improvement.
("Simple" for someone who knows the code base enough)

**Comment by Calinou** [comment]
What about the window *size*? Should it also be remembered across runs?

**Comment by ngreve** [comment]
Wouldn't it be possible to recycle an already existing run window?
By not closing the old run window and not creating a new one, but rather reuse the existing one?
Just a thought, I do not know how the windowing system works under the hood.

**Comment by Calinou** [comment]
> Wouldn't it be possible to recycle an already existing run window?
By not closing the old run window and not creating a new one, but rather reuse the existing one?

Restarting the project will close a process and create a new one, so this isn't possible.

**Comment by DanielNautre** [comment]
> Unfortunately, this caveat can also make the feature confusing: changes to Test Width/Test Height will not apply anymore to your project if you've run the project at least once before.

Woudn't that (change to test height/width) constitute an acceptable case to reset the position/size of the run window ?

**Comment by Sosasees** [comment]
yes, i think it would, because i change the Test Size only a few times at the beginning of development of a new project and then don't change it again.

**Comment by Nallebeorn** [comment]
Could be useful to know that you can work around this with a small editor plugin that writes to the editor settings whenever the window is moved. You can't access `EditorInterface.get_editor_settings` directly from the running game, but you can circumvent this by writing to a file or sending the data over the debugger.

I made an example here: https://gist.github.com/Nallebeorn/348add66c16a6cf76857e91a0202967e.

**Comment by discordier** [comment]
> Could be useful to know that you can work around this with a small editor plugin that writes to the editor settings whenever the window is moved. You can't access `EditorInterface.get_editor_settings` directly from the running game, but you can circumvent this by writing to a file or sending the data over the debugger.
> 
> I made an example here: https://gist.github.com/Nallebeorn/348add66c16a6cf76857e91a0202967e.

Thanks a bunch for this, I really love it and added your plugin to my tool belt (empty boilerplate Godot project). Never want to miss it again.

You SHOULD add it to the asset store.

**Comment by Nallebeorn** [comment]
> Thanks a bunch for this, I really love it and added your plugin to my tool belt (empty boilerplate Godot project). Never want to miss it again.
> 
> You SHOULD add it to the asset store.

It can now be found at https://store.godotengine.org/asset/nallebeorn/remember-window-placement/ and https://godotengine.org/asset-library/asset/5365 for anyone interested.
